process_test.go

```diff
@@ -408,6 +408,115 @@ func TestChannelNotFound(t *testing.T) {
 	}
 }
 
+// TestOutgoingChannelNotFound tests the case where the outgoing channel for a htlc is
+// not found in two cases:
+// 1. The HTLC was settled: the channel must exist, so we fail if it's not found
+// 2. The HTLC was failed: the outgoing channel could be bogus, so we handle the error
+func TestOutgoingChannelNotFound(t *testing.T) {
+	tests := []struct {
+		name          string
+		settled       bool
+		outgoingFound bool
+		err           error
+	}{
+		{
+			name:          "outgoing found, settled",
+			settled:       true,
+			outgoingFound: true,
+		},
+		{
+			name:          "outgoing found, not settled",
+			settled:       false,
+			outgoingFound: true,
+		},
+		{
+			name:          "outgoing not found, settled",
+			settled:       true,
+			outgoingFound: false,
+			err:           errChannelNotFound,
```

Comment:
Yes we still don't account for pending channels because the pending API doesn't surface all the info we need. 

This has always existed for the incoming channel and it doesn't seem like anybody has run into issues? But we could downgrade to just log whenever channel is not found to be cautious?

Reply:
That was also what I thought. But agreed that it seems far off the hot path. Whatever you think is best.